geomstats/geometry/discrete_curves.py

```diff
@@ -1229,17 +1256,45 @@ class SRVMetric(ElasticMetric):
     """
 
     def __init__(
-        self, ambient_manifold, ambient_metric=None, translation_invariant=True
+        self,
+        ambient_manifold,
+        ambient_metric=None,
+        translation_invariant=True,
+        k_sampling_points=10,
     ):
+        self.ambient_manifold = ambient_manifold
         super().__init__(
-            a=1,
-            b=0.5,
-            ambient_manifold=ambient_manifold,
-            ambient_metric=ambient_metric,
-            translation_invariant=translation_invariant,
+            dim=ambient_manifold.dim,
+            shape=(k_sampling_points,) + ambient_manifold.shape,
         )
+        self.ambient_metric = ambient_metric
+        if ambient_metric is None:
+            if hasattr(ambient_manifold, "metric"):
+                self.ambient_metric = ambient_manifold.metric
+            else:
+                raise ValueError(
+                    "Instantiating an object of class "
+                    "ElasticMetric requires either a metric"
+                    " or an ambient manifold"
+                    " equipped with a metric."
+                )
+        self.l2_curves_metric = L2CurvesMetric(ambient_manifold=ambient_manifold)
```

Comment:
can be deleted since it is defined in `define_embedding_metric`. then in the code use `self.embedding_metric`.

Reply:
Yes you're right, much better, thanks !